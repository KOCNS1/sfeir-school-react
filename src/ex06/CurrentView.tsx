import {Loading} from "../solution/Loading";
import {Carousel} from "../ex03/Carousel";
import React from "react";
import {PersonCard} from "../ex02/PersonCard";

type CurrentViewProps = {
  people: People;
  showList: boolean;
};

export const CurrentView: React.FC<CurrentViewProps> = ({people, showList}) => {
  const list = people.map(person => <PersonCard key={person.id} person={person} />);
  const carousel = <Carousel people={people} />;
  const view = showList ? list : carousel;

  return (
      <>
        {people.length === 0 ? <Loading/> : view}
      </>
  );
};