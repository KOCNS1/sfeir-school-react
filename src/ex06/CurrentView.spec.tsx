import {CurrentView} from "./CurrentView";
import React from "react";
import renderer from "react-test-renderer";

describe("CurrentView component", () => {

  it("should render Loading", () => {
    const rendered = renderer.create(<CurrentView people={[]} showList={true} />);

    expect(rendered.toJSON()).toMatchSnapshot();
  });

});

