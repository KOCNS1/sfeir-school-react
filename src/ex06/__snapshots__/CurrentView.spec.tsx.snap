// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CurrentView component should render Carousel 1`] = `
<div
  className="flex-row"
>
  <button
    className="mdc-fab mdc-fab--mini"
    onClick={[Function]}
  >
    <i
      className="rmwc-icon material-icons mdc-fab__icon"
    >
      skip_previous
    </i>
  </button>
  <div
    className="carousel"
  >
    <section
      className="current mdc-card"
      id="p-123"
    >
      <div
        className="card-content content-type-person-info"
      >
        <figure>
          <img
            alt="face of Jean-Michel"
            src="string"
          />
        </figure>
        <header>
          <h1
            className="mdc-typography--headline5"
          >
            <a
              href="#/person/123"
            >
              Jean-Michel Test
            </a>
          </h1>
          <h2
            className="mdc-typography--subtitle1"
          >
            string
          </h2>
        </header>
        <p>
          <i
            className="rmwc-icon material-icons rmwc-icon--size-small"
            title="email"
          >
            email
          </i>
           
          <a
            href="mailto:string"
          >
            string
          </a>
        </p>
        <p>
          <i
            className="rmwc-icon material-icons rmwc-icon--size-small"
            title="phone"
          >
            phone
          </i>
           
          <a
            href="tel:string"
          >
            string
          </a>
        </p>
        <p>
          <i
            className="rmwc-icon material-icons rmwc-icon--size-small"
            title="manager"
          >
            supervisor_account
          </i>
           
          <a
            href="#/person/string"
          >
            string
          </a>
        </p>
      </div>
    </section>
  </div>
  <button
    className="mdc-fab mdc-fab--mini"
    onClick={[Function]}
  >
    <i
      className="rmwc-icon material-icons mdc-fab__icon"
    >
      skip_next
    </i>
  </button>
</div>
`;

exports[`CurrentView component should render List 1`] = `
<section
  className="mdc-card"
>
  <div
    className="card-content content-type-person-info"
  >
    <figure>
      <img
        alt="face of Jean-Michel"
        src="string"
      />
    </figure>
    <header>
      <h1
        className="mdc-typography--headline5"
      >
        <a
          href="/person/123"
        >
          Jean-Michel
           
          Test
        </a>
      </h1>
      <h2
        className="mdc-typography--subtitle1"
      >
        string
      </h2>
    </header>
    <p>
      <i
        className="rmwc-icon material-icons rmwc-icon--size-small"
      >
        email
      </i>
       
      <a
        href="mailto:string"
      >
        string
      </a>
    </p>
    <p>
      <i
        className="rmwc-icon material-icons rmwc-icon--size-small"
      >
        phone
      </i>
       
      <a
        href="tel:string"
      >
        string
      </a>
    </p>
    <div>
      hello
    </div>
  </div>
</section>
`;

exports[`CurrentView component should render Loading 1`] = `
<main>
  <div
    _size="xlarge"
    className="rmwc-circular-progress rmwc-circular-progress--size-xlarge rmwc-circular-progress--indeterminate rmwc-circular-progress--thickerstroke"
  >
    <svg
      className="rmwc-circular-progress__circle"
      viewBox="0 0 48 48"
    >
      <circle
        className="rmwc-circular-progress__path"
        cx={24}
        cy={24}
        r={20}
      />
    </svg>
  </div>
</main>
`;
